src/snowflake/snowpark/mock/mock_select_statement.py

```diff
@@ -522,3 +532,56 @@ def sql_query(self) -> str:
     @property
     def schema_query(self) -> str:
         return self.snowflake_plan.schema_query
+
+
+class MockJoinStatement(MockSelectable):
+    def __init__(
+        self,
+        left: Selectable,
+        right: Selectable,
+        analyzer: Optional["Analyzer"] = None,
+        on=None,
+        how="left",
+        lsuffix="",
+        rsuffix="",
+    ) -> None:
+        super().__init__(analyzer=analyzer)
+
+        self.left = left
+        self.right = right
+        self.on = on
+        self.how = how
+        self.api_calls = None
+
+        for operand in (
+            left,
+            right,
+        ):
+            if operand.pre_actions:
```

Comment:
This block is probably not needed since there's no pre_actions or post_actions required.